app/models/daily_summary.rb

```diff
@@ -0,0 +1,81 @@
+#  impressions_count                   :integer          default(0), not null
+#  clicks_count                        :integer          default(0), not null
+#  click_rate                          :decimal(, )      default(0.0), not null
+#  ecpm_cents                          :integer          default(0), not null
+#  ecpm_currency                       :string           default("USD"), not null
+#  estimated_gross_revenue_cents       :integer          default(0), not null
+#  estimated_gross_revenue_currency    :string           default("USD"), not null
+#  estimated_property_revenue_cents    :integer          default(0), not null
+#  estimated_property_revenue_currency :string           default("USD"), not null
+#  estimated_house_revenue_cents       :integer          default(0), not null
+#  estimated_house_revenue_currency    :string           default("USD"), not null
+#  displayed_at_date                   :date             not null
+#  created_at                          :datetime         not null
+#  updated_at                          :datetime         not null
+#
+
+class DailySummary < ApplicationRecord
+  # extends ...................................................................
+  # includes ..................................................................
+
+  # relationships .............................................................
+  belongs_to :impressionable, polymorphic: true
+
+  # validations ...............................................................
+  validates :impressions_count, numericality: {greater_than_or_equal_to: 0}
+  validates :clicks_count, numericality: {greater_than_or_equal_to: 0}
+  validates :click_rate, numericality: {greater_than_or_equal_to: 0}
+  validates :displayed_at_date, presence: true
+
+  # callbacks .................................................................
+  before_save :set_click_rate
+  before_save :set_ecpm
+
+  # scopes ....................................................................
+  scope :on, ->(*dates) { where displayed_at_date: dates.map { |date| Date.coerce(date) } }
+  scope :between, ->(start_date, end_date = nil) {
+    where displayed_at_date: Date.coerce(start_date)..Date.coerce(end_date)
+  }
+  scope :scoped_by, ->(record) {
+    record ? where(scoped_by_type: record.class.name, scoped_by_id: record.id) : all
+  }
+
+  # additional config (i.e. accepts_nested_attribute_for etc...) ..............
+  monetize :ecpm_cents, numericality: {greater_than_or_equal_to: 0}
+  monetize :estimated_gross_revenue_cents, numericality: {greater_than_or_equal_to: 0}
+  monetize :estimated_property_revenue_cents, numericality: {greater_than_or_equal_to: 0}
+  monetize :estimated_house_revenue_cents, numericality: {greater_than_or_equal_to: 0}
+
+  # class methods .............................................................
+  class << self
+  end
+
+  # public instance methods ...................................................
+
+  def impressions_per_mille
```

Comment:
I'm not sure this name or method makes sense in terms of advertising. Is this method necessary? It's just caused me confusion a couple of times now.

Reply:
It's necessary to calculate eCPM but I'll move it to be an internal implementation detail of `set_ecpm`.